server/boot/user.js

```diff
@@ -56,6 +55,61 @@ function encodeFcc(value = '') {
   return replaceScriptTags(replaceFormAction(value));
 }
 
+function isAlgorithm(challenge) {
+  // test if name starts with hike/waypoint/basejump/zipline
+  // fix for bug that saved different challenges with incorrect
+  // challenge types
+  return !(/^(waypoint|hike|zipline|basejump)/i).test(challenge.name) &&
```

Comment:
The migration tool will have to be written, I presume, and it's trivial to add this as a feature.
`challenge.title = challenge.title.replace(/^(waypoint|hike|zipline|basejump|checkpoint):\s*/i, "");`